src/main/kotlin/org/rust/cargo/toolchain/Cargo.kt

```diff
@@ -45,6 +45,16 @@ class Cargo(
     private val cargoExecutable: Path,
     private val rustExecutable: Path
 ) {
+    val supportsBuildCheckAllTargets: Boolean = checkSupportForBuildCheckAllTargets()
```

Comment:
Let's use `checkSupportForBuildCheckAllTargets` directly where we need this info to avoid unnecessary invocation of `checkSupportForBuildCheckAllTargets` for every `Cargo` object creation